src/test/java/uk/ac/ebi/eva/contigalias/controller/admin/AdminHandlerTest.java

```diff
@@ -14,72 +14,65 @@
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
 
 import uk.ac.ebi.eva.contigalias.entities.AssemblyEntity;
 import uk.ac.ebi.eva.contigalias.entitygenerator.AssemblyGenerator;
 import uk.ac.ebi.eva.contigalias.service.AssemblyService;
 
 import java.io.IOException;
-import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.mock;
-import static uk.ac.ebi.eva.contigalias.controller.BaseController.DEFAULT_PAGE_NUMBER;
-import static uk.ac.ebi.eva.contigalias.controller.BaseController.DEFAULT_PAGE_SIZE;
 
-public class AdminControllerTest {
+public class AdminHandlerTest {
 
     private final AssemblyEntity entity = AssemblyGenerator.generate();
 
-    private AdminController controller;
+    private AdminHandler handler;
 
     @BeforeEach
     void setup() throws IOException {
         AssemblyService mockAssemblyService = mock(AssemblyService.class);
-        List<AssemblyEntity> entityAsList = Collections.singletonList(entity);
+        Optional<AssemblyEntity> entityAsList = Optional.of(entity);
         Mockito.when(mockAssemblyService.getAssemblyOrFetchByAccession(entity.getGenbank()))
                .thenReturn(entityAsList);
-        Mockito.when(mockAssemblyService
-                             .getAssemblyOrFetchByAccession(entity.getRefseq()))
+        Mockito.when(mockAssemblyService.getAssemblyOrFetchByAccession(entity.getRefseq()))
                .thenReturn(entityAsList);
-        controller = new AdminController(mockAssemblyService);
+        handler = new AdminHandler(mockAssemblyService);
     }
 
     @Test
     public void getAssemblyOrFetchByAccessionGCA() throws IOException {
-        ResponseEntity<List<AssemblyEntity>> assemblyByAccession =
-                controller.getAssemblyOrFetchByAccession(entity.getGenbank(), DEFAULT_PAGE_NUMBER, DEFAULT_PAGE_SIZE);
-        testAssemblyEntityResponse(assemblyByAccession);
+        List<AssemblyEntity> assemblyByAccession = handler.getAssemblyOrFetchByAccession(entity.getGenbank());
+        assertFirstIdenticalToEntity(assemblyByAccession);
     }
 
     @Test
     public void getAssemblyOrFetchByAccessionGCF() throws IOException {
-        ResponseEntity<List<AssemblyEntity>> assemblyByAccession =
-                controller.getAssemblyOrFetchByAccession(entity.getRefseq(), DEFAULT_PAGE_NUMBER, DEFAULT_PAGE_SIZE);
-        testAssemblyEntityResponse(assemblyByAccession);
+        List<AssemblyEntity> assemblyByAccession = handler.getAssemblyOrFetchByAccession(entity.getRefseq());
+        assertFirstIdenticalToEntity(assemblyByAccession);
     }
 
     @Test
     public void test404NotFound() throws IOException {
```

Comment:
maybe this shouldn't be called 404 if it doesn't deal with HTTP codes. The test in the controller test is called test404NotFound too, but that one is fine.